lib/Backends/BackendUtils.cpp

```diff
@@ -55,3 +55,105 @@ size_t glow::getValueOffset(
   assert(it != symbolTable.end() && "Symbol not found.");
   return it->second.offset;
 }
+
+runtime::RuntimeBundle
+glow::generateRuntimeBundle(const IRFunction *F,
+                            MemoryAllocator *constantAllocator,
+                            MemoryAllocator *placeholderAllocator,
+                            MemoryAllocator *activationsAllocator) {
+  // Handle Constants, Placeholders, and Activations, in that order.
+  /// Symbol table mapping symbol name to offset for runtime.
+  std::unordered_map<std::string, runtime::RuntimeSymbolInfo> symbolTable;
+  uint64_t constantMaxSize = 0;
+  uint64_t placeholderMaxSize = 0;
+  uint64_t activationsMaxSize = 0;
+
+  // Compute the offsets for Constants.
+  if (constantAllocator) {
+    for (auto &v : F->getGraph()->getParent()->getConstants()) {
+      assert(isa<WeightVar>(F->getWeightForNode(v)) && "Expected WeightVar");
+      auto *w = cast<WeightVar>(F->getWeightForNode(v));
+      auto numBytes = w->getSizeInBytes();
+      size_t addr = constantAllocator->allocate(numBytes, v);
+      runtime::RuntimeSymbolInfo symbol;
+      symbol.size = numBytes;
+      symbol.offset = addr;
+      symbol.type = *w->getType();
+      symbolTable.emplace(std::string(v->getName()), symbol);
+    }
+    constantMaxSize = constantAllocator->getMaxMemoryUsage();
+  }
```

Comment:
What happen if the function has constants and we haven't provided a memory allocator?

Reply:
We don't do allocation for constants if a constant allocator is not provided.

Reply:
At the very least we should report an error if we don’t provide a constant allocator whereas there are some constants in the function.

Another thing is do we need the flexibility of having those allocators optional?

Reply:
Probably not, @opti-mix and I were discussing this as well. I think they will be required references. It it doesn't get used that's fine.

Reply:
Agree